dashboard-app/src/util/hooks/useRequestOnLoad.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react';
+import { AxiosPromise, AxiosResponse } from 'axios';
+import { Response } from '../response';
+
+export default (apiCall: AxiosPromise) => {
```

Comment:
Is it better to make this eager or lazy? By which I mean, if you make the parameter a promise, it means the network request will already be happening by the time we get inside this hook. Is that OK? Or do we want to force the network request not to happen until we're inside `useEffect`?

Reply:
Yeah i was thinking about this. If we want it to be lazy we'll need to refactor how we are defining our parameters and passing it on. It will most likely look like an object that will return a bunch of custom hooks.

eg
```js
const CbAdmin = {
	getLogs: (params, updateOn = []) => {
		return useRequest(params, updateOn, `v1/users/volunteer-logs`)
	} 
}
```

I wanted to hold off on this refactor until i had a better feeling of hooks, to see how best to implement. Specifically how we want the abstraction to define how frequently the call is made (shown above with `updateOn`).